### PR TITLE
Porting Audio to C++

### PR DESCRIPTION
**Pull request by wdigger**
Already familiar sequence of porting.

### CODE REVIEW

**Comment by wdigger** [comment]
Any comments before merge?

File: src/audio-sdlmixer.cc [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+}
+
+/*
+static void
+midi_track_finished() {
+  if (midi_enabled) {
+    midi_play_track(current_track);
+  }
+}
+*/
```
Comment (jonls) [inline src/audio-sdlmixer.cc]: This code is commented out. Is it not needed any more?

Comment (wdigger) [inline src/audio-sdlmixer.cc]: It's my fault. I decided that porting as is impractical, but forgot to do well (:

**Comment by wdigger** [comment]
Continuous music playback restored.

File: src/event_loop.h [hunk 2]
```diff
@@ -57,6 +57,11 @@ class event_handler_t {
 
 typedef std::list<event_handler_t*> event_handlers_t;
 
+class deferred_callee_t {
+ public:
+  virtual void deffered_call(void *data) = 0;
```
Comment (jonls) [inline src/event_loop.h]: `deffered_call` -> `deferred_call`

**Comment by jonls** [comment]
Code looks good. The comment block with `midi_track_finished ` is still in there. Can it be removed now?

**Comment by wdigger** [comment]
I remove commented code, and can merge all.

**Comment by jonls** [comment]
:+1:

**Comment by zdechlak** [comment]
Hi, I am Zdechlak (beta-tester freeserf game) Hello! I would like to ask if
you could send me the newest code compilation for Windows - I would check
the game development progress :)

**Comment by jonls** [comment]
@zdechlak https://ci.appveyor.com/project/jonls/freeserf/build/55/job/c6t83j87bamf5m87/artifacts You can expand the box with the checks and follow the link to the AppVeyor page for any pull request.

**Comment by zdechlak** [comment]
i have sxtrace.exe problem :P

[image: Obraz w treści 1]




2016-01-20 1:01 GMT+01:00 Jon Lund Steffensen <notifications@github.com>:

**Comment by wdigger** [comment]
@zdechlak try to download and run release version
https://ci.appveyor.com/project/jonls/freeserf/build/55/job/224m6ut1x4d7g2hi/artifacts

**Comment by zdechlak** [comment]
hi wdigger, your @ ?

2016-01-20 10:02 GMT+01:00 wdigger <notifications@github.com>:

**Comment by wdigger** [comment]
@zdechlak you are welcome, wicked_digger@mail.ru

**Pull request merged**